packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -759,6 +763,87 @@ export class RpcContractProvider extends Provider implements ContractProvider, S
 
     return batch;
   }
+
+  /**
+   *
+   * @description Stake the given amount for the source. The source must be a delegator to be allowed to stake
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param stake operation parameter
+   *
+   * @throws {@link InvalidAmountError}
+   */
+  async stake(params: StakeParams) {
+    validateStakingParams(params, 'stake');
+    const estimate = await this.estimate(params, this.estimator.stake.bind(this.estimator));
+    const publicKeyHash = await this.signer.publicKeyHash();
+    const source = params.source ?? publicKeyHash;
+
+    const prepared = await this.prepare.stake({ ...params, ...estimate });
+    const content = prepared.opOb.contents.find(
+      (op) => op.kind === OpKind.TRANSACTION
+    ) as OperationContentsTransaction;
+
+    const opBytes = await this.forge(prepared);
+    const { hash, context, forgedBytes, opResponse } = await this.signAndInject(opBytes);
+    return new TransactionOperation(hash, content, source, forgedBytes, opResponse, context);
+  }
+
+  /**
+   *
+   * @description Unstake the given amount. If "everything" is given as amount, unstakes everything from the staking balance. Unstaked tez remains frozen for a set amount of cycles (the slashing period) after the operation. Once this period is over, the operation "finalize unstake" must be called for the funds to appear in the liquid balance.
```

Comment:
I understand it's the client's description, might need to take out `If "everything" is given as amount, unstakes everything from the staking balance.`. 
UnstakeParams.amount is type number, for now we don't want to take string 'everything' and query the staked balance and used it as  TransactionOperationParameter.value